pkg/blobstore/cloud/cloud_blob_access.go

```diff
@@ -17,18 +19,20 @@ import (
 type cloudBlobAccess struct {
 	bucket            *blob.Bucket
 	keyPrefix         string
-	readBufferFactory ReadBufferFactory
+	readBufferFactory blobstore.ReadBufferFactory
 	digestKeyFormat   digest.KeyFormat
+	beforeCopy        BeforeCopyFunc
 }
 
 // NewCloudBlobAccess creates a BlobAccess that uses a cloud-based blob storage
 // as a backend.
-func NewCloudBlobAccess(bucket *blob.Bucket, keyPrefix string, readBufferFactory ReadBufferFactory, digestKeyFormat digest.KeyFormat) BlobAccess {
+func NewCloudBlobAccess(bucket *blob.Bucket, keyPrefix string, readBufferFactory blobstore.ReadBufferFactory, digestKeyFormat digest.KeyFormat, beforeCopy BeforeCopyFunc) blobstore.BlobAccess {
 	return &cloudBlobAccess{
 		bucket:            bucket,
 		keyPrefix:         keyPrefix,
 		readBufferFactory: readBufferFactory,
 		digestKeyFormat:   digestKeyFormat,
+		beforeCopy:        beforeCopy,
 	}
 }
 
```

Comment:
This solution only triggers copies when doing FindMissing(). Though this may be acceptable, it's important document somewhere that it only 'fixes' the CAS. Using this for the AC still causes actions to get lost, even when accessed. This is different from, say, LocalBlobAccess which touches objects both in Get() and FindMissing().

Out of interest, can we think of any ways to also fix this for Get()? Is there anything better we can do than sending two separate calls (one GetObject, one CopyObject)?

Reply:
Good point, this isn't really useful if we don't also have AC working.

It'll have to be two separate API calls on the wire - one GET to retrieve the content because we need that anyway, and one COPY to update the metadata. Given that, the necessary change is straightforward, I'll sort it out.

Reply:
Duly fixed - it's easy because this COPY can be entirely async